### PR TITLE
Rspec controller

### PR DESCRIPTION
**Pull request by sajibarafatsiddiqui**
- Added RSpec as a gem to the project.
- Created a Request spec file for all the controllers.
- Checked If the response status was correct.
- Checked If a correct template was rendered.
 - Checked If the response body includes the correct placeholder text.
Thanks

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -1,7 +1,44 @@
...
-RSpec.describe 'Posts', type: :request do
-  describe 'GET /index' do
-    pending "add some examples (or delete) #{__FILE__}"
+RSpec.describe PostsController, type: :controller do
+  before { get :index, params: { user_id: 1 } }
+  it { should respond_with(200) }
+end
+
+RSpec.describe PostsController, type: :controller do
+  it { should route(:get, 'users/1/posts').to(action: :index, user_id: 1) }
```
Comment (BenMukebo) [inline spec/requests/posts_spec.rb]: - [ ] Great job 👏🏻 However, you should use `type: request` as shown in the example code OR use `config.infer_spec_type_from_file_location!` in the Rspec config file (in that case just placing a file in the `spec/requests` dir will be enough).  `config.infer_spec_type_from_file_location!` is a configuration setting that allows RSpec to automatically infer the type of a spec file based on its location in the directory structure.
- So, to ensure that a spec file is treated as a request spec, you can either explicitly specify the type of the spec using type: :request in the spec file itself, like so:

   ```ruby
    require 'rails_helper'
   
    RSpec.describe "Users", type: :request do
      ...
    end
   ```
- Make sure. `index` and `show` actions for `User` and `Post` controllers are tested

   ```ruby
   describe 'GET /index' do
     ...
   end
   
   describe 'GET /show' do
     ...
   end
   ```

- You could make reference to this [PR](https://github.com/Nkaleth/blog_web/pull/6) if you have any question.

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @sajibarafatsiddiqui,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used.  ✔️   
- Good commit messages ✔️ 
- Professional README ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (sajibarafatsiddiqui) [reply spec/requests/posts_spec.rb]: Sorry, I missed that point. Thanks for pointing out. Corrected accordingly.

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @sajibarafatsiddiqui          , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] linters are passing
- [x] Professional README 💯
- [x] Controller specs are passing🥇  

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Review by Gambit142 — APPROVED** [review]
Hi @sajibarafatsiddiqui 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- All linters checks passed ✔️ 
- Use of `Rspec` as testing library ✔️ 
- Wrote request specs for `Post` and `User` models ✔️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**